src/pytorch_lightning/utilities/data.py

```diff
@@ -475,45 +503,82 @@ def wrapper(obj: Any, *args: Any, **kwargs: Any) -> None:
         }
 
         if not hasattr(obj, "__pl_saved_args"):
-            obj.__pl_saved_args = args
-            obj.__pl_saved_kwargs = kwargs
-            obj.__pl_saved_arg_names = param_names
-            obj.__pl_saved_default_kwargs = default_kwargs
+            object.__setattr__(obj, "__pl_saved_args", args)
+            object.__setattr__(obj, "__pl_saved_kwargs", kwargs)
+            object.__setattr__(obj, "__pl_saved_arg_names", param_names)
+            object.__setattr__(obj, "__pl_saved_default_kwargs", default_kwargs)
```

Comment:
do we need to call this ecplicitly? can't we do `setattr(obj, "__pl_saved_args", args)`

Reply:
I'd almost say that's a philosophical question :joy:

If for some reason a user overwrites their `__setattr__` (for example to mimic frozen class), then calling normal `setattr(obj, ..., ...)` would not work. So, that and running into `RecursionError`s while testing were the reasons why I decided to use `object.__setattr__(obj, ..., ...)` everywhere in the case of these internal attributes

Reply:
Makes sense, I was just wondering if it was an issue with class binding or whether this was for other reasons :) Because I actually prefer not to call magic methods directly if possible, but in this case it makes sense :)